renku_notebooks/api/schemas.py

```diff
@@ -130,10 +130,10 @@ class UserPodAnnotations(
     class Meta:
         unknown = INCLUDE
 
-    def get_attribute(self, obj, key, *args, **kwargs):
+    def get_attribute(self, obj, key, default, *args, **kwargs):
```

Comment:
where is `default` taken from?

Reply:
@rokroskar default is passed by marshmallow depending on how the schema is specified. If the schema allows for missing values and it has defaults set then it will be used. I.e. https://marshmallow.readthedocs.io/en/stable/marshmallow.fields.html#marshmallow.fields.String if a default value is set then that value will be used.

Reply:
Before what I had was just `obj[key]`. And this was erroring out because `git-host` was not marked as required in the schema and marshmallow expects you to have something that allows using a default value that will not error out (https://marshmallow.readthedocs.io/en/stable/api_reference.html#marshmallow.Schema.get_attribute). The signature of the function in the docs specifically has the default value.

For a missing key that is not required marshmallow has some default default value that it uses and then it fully removes this field from the results. It makes sense since the field was not required and it has no default value it should not be shown at all in the results.

Reply:
Also in the docs marshmallow has an example of overriding the `get_attribute` method in the same way as what I have in my fix here.

Reply:
https://marshmallow.readthedocs.io/en/stable/extending.html#overriding-how-attributes-are-accessed

Reply:
ok great, that makes sense to me